flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/AbstractRestHandler.java

```diff
@@ -111,10 +112,13 @@ protected AbstractRestHandler(
 	private void processRestHandlerException(ChannelHandlerContext ctx, HttpRequest httpRequest, RestHandlerException rhe) {
 		log.error("Exception occurred in REST handler.", rhe);
 
+		String serverSideExceptionMesssage = "\n<Exception on server side:\n" +
+			ExceptionUtils.stringifyException(rhe) + "\nEnd of exception on server side>";
+
 		HandlerUtils.sendErrorResponse(
 			ctx,
 			httpRequest,
-			new ErrorResponseBody(rhe.getMessage()),
+			new ErrorResponseBody(Arrays.asList(rhe.getMessage(), serverSideExceptionMesssage)),
```

Comment:
I would prefer if we only did this in case of an Internal server error as a fallback.

Reply:
We also may have to double-check that the list of errors is properly handled in all places, like the client and WebUI.